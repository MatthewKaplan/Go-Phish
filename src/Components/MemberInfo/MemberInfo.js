import React, { Component } from "react";
import "./MemberInfo.scss";

class MemberInfo extends Component {
  render() {
    const { currentMember } = this.props;
    return (
      <div className="member-info-popup">
        <div className="member-info">
          <div
            className="close-popup"
            onClick={() => this.props.closePopup()}
          />
          <section className="info">
            <h1>{currentMember.member}</h1>
            <h4>{currentMember.instruments}</h4>
            <section className="member-body">
              <h3>Also known as:</h3>
              <p>{currentMember.also_known_as}</p>
              <br />
              <h3>Side Projects:</h3>
              <p>{currentMember.side_projects}</p>
            </section>
            <br />
            <section className="member-footer">
              <h3>Born:</h3>
              <h4>{currentMember.date_of_birth}</h4>
              <h4>{currentMember.born}</h4>
            </section>
          </section>
        </div>
      </div>
    );
  }
}

export default MemberInfo;
