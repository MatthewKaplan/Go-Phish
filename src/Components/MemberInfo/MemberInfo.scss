.member-info-popup {
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  background-color: rgba(0,0,0, 0.5);
  z-index: 2;
}

.member-info {
  position: absolute;
  left: 25%;
  right: 25%;
  top: 25%;
  bottom: 25%;
  height: 600px;
  margin: 0 auto;
  background-image: url('https://i.imgur.com/9GB0xeq.jpg');
  background-repeat: no-repeat;
  background-size: contain;
  border-radius: 12px;
}

.info {
  width: 80%;
  margin: 105px auto;
  background-color: rgba(0, 0, 0, 0.706);
  color: #fff;
  height: 300px;
  padding: 10px;
  border-radius: 12px;
}

.member-body {
  text-align: left;
}

h3 {
  color: #C7142D;
  -webkit-text-stroke-width: .2px;
  -webkit-text-stroke-color: #004D95;
}

.member-footer {
  text-align: right;
}

.close-popup {
  background-image: url('../../Assets/icon.svg');
  background-repeat: no-repeat;
  background-size: contain;
  height: 60px;
  width: 60px;
  float: right;
  cursor: pointer;
  margin: 10px 10px 0 0;

  &:hover {
    opacity: .8;
  }
}