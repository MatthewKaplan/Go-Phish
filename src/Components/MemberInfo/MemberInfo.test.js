import React from "react";
import { MemberInfo } from "./MemberInfo";
import MockData from "../../Helpers/mockData";
import { shallow } from "enzyme";

const member = MockData.mockMember;

describe("MemberInfo", () => {
  let wrapper, instance, mockClosePopup;

  beforeEach(() => {
    mockClosePopup = jest.fn();
    wrapper = shallow(
      <MemberInfo currentMember={member} closePopup={mockClosePopup} />
    );
    instance = wrapper.instance();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should invoke 'closePopup' when button is clicked", () => {
    wrapper.find("[data-test='close-popup']").simulate("click");
    expect(mockClosePopup).toHaveBeenCalled();
  });
});
