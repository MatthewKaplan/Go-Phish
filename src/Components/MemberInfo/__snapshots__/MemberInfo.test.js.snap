// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberInfo should match the snapshot 1`] = `ShallowWrapper {}`;
